### DISCUSSION TITLE
Dynamic ERB cassettes could support reverse substitution

### DISCUSSION THREAD

**Discussion by jkremser** [vcr/vcr#874] [Ideas]
This is an enhancement.

By "reverse substitution" I mean this:
if the VCR is recording with those options `erb: {my_foo_uuid: '28026b36-8fe4-4332-84c8-524e173a68bf'}` and if the `28026b36-8fe4-4332-84c8-524e173a68bf` is somewhere in the data that is about to save in the cassette, instead of this value store the key `<%= my_foo_uuid %>`.

It probably should not be enabled by default, because if options are `erb: {foo: 1}`, it'd be bad to replace every `1` with `<%= foo %>`, but you got the idea. So some additional option, like `reverse_substitution: true` could be added to turn this feature on.

Why to do that? Dynamic ERB cassettes is a nice feature, but it needs to be hand written, no automation here. This way, they can be recorded.

I am willing to help with this if there is demand for this feature.

**Comment by krainboltgreene** [comment]
This doesn't actually need code, you can write a reader to use erb + yaml.

**Comment by jkremser** [comment]
Could you please elaborate? I can't find anything like a "reader", do you mean this thing from version 2.0.0 https://docs.omniref.com/ruby/gems/vcr/2.0.0/files/lib/vcr/cassette/reader.rb ? this looks more responsible for reading the cassettes than persisting them into file system. I guess, I could duck-type the `file_content` method there and do some magic in it, but the real issue for us is more constantly rewriting (in git) the cassettes with slightly different data in it (that changes could have been 'templated').

Or is it better to not put those cassettes into git repo?

**Comment by jkremser** [comment]
btw. this looks related to https://github.com/vcr/vcr/issues/6
